sympy/simplify/tests/test_simplify.py

```diff
@@ -580,7 +580,8 @@ def test_besselsimp():
     assert besselsimp(sqrt(2)*sqrt(pi)*x**(S(1)/4)*exp(I*pi/4)*exp(-I*pi*a/2) *
                       besseli(-S(1)/2, sqrt(x)*exp_polar(I*pi/2)) *
                       besseli(a, sqrt(x)*exp_polar(I*pi/2))/2) == \
-        besselj(a, sqrt(x)) * cos(sqrt(x))
+        sqrt(2)*(1 + I)*exp_polar(-I*pi/4)*exp(-I*pi*a/2) *
```

Comment:
It seems that `besselsimp` is broken by the change in `exp.eval`. That should be investigated and fixed before this PR can be merged.